### PR TITLE
Add script for TDFN-14_1EP

### PR DESCRIPTION
**Pull request by drid**
Based on:
http://pdfserv.maximintegrated.com/package_dwgs/21-0137.PDF
http://pdfserv.maximintegrated.com/land_patterns/90-0063.PDF

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
   suffix: ''
   #include_suffix_in_3dpath: 'False'
   #include_pad_size: 'fp_name_only' # 'both' | 'none' (default)
+
+TDFN-14-1EP_3x3mm_P0.4mm_EP1.78x2.35mm:
+  device_type: 'TDFN'
+  size_source: 'https://pdfserv.maximintegrated.com/land_patterns/90-0063.PDF'
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:693]: Please append T1433-2C

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+  body_size_x: 3 +/-0.1
+  body_size_y: 3 +/-0.1
+  overall_height: 0.7
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:696]: Should be 0.7 .. 0.75 .. 0.8

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 3]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+  lead_width: 0.2 +/-0.02
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:697]: Tolerance is 0.05

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 4]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+  lead_len: 0.7 +/-0.02
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:698]: 0.2 .. 0.3 .. 0.4

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 5]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+
+  EP_size_x: 1.78 +/-0.02
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:700]: 1.7 +/- 0.1

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 6]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+  EP_size_y: 2.35 +/-0.02
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:701]: 2.3 +/-0.1

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 7]
```diff
@@ -687,3 +687,39 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+
+  thermal_vias:
+    count: [2, 3]
+    drill: 0.2
+    # min_annular_ring: 0.15
+    paste_via_clearance: 0.1
+    EP_num_paste_pads: [2, 2]
+    #paste_between_vias: 1
+    #paste_rings_outside: 1
+    EP_paste_coverage: 0.75
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:714]: Paste coverage is on the higher end.
Does Maxim recommend 75% somewhere?

Comment (drid) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:714]: Based on QFN its 50% - 80% (https://www.maximintegrated.com/en/design/technical-documents/app-notes/6/6382.html) 
I guess I could go 65%?

Comment (chschlue) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:714]: If there's no particular reason for a special value, please remove the line entirely.
That way, the generator can take care of it (and might be updated if a better default than 65% comes up)

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 8]
```diff
@@ -773,6 +772,44 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
   #include_suffix_in_3dpath: 'False'
   #include_pad_size: 'fp_name_only' # 'both' | 'none' (default)
 
+TDFN-14-1EP_3x3mm_P0.4mm_EP1.78x2.35mm:
+  device_type: 'TDFN'
+  size_source: 'https://pdfserv.maximintegrated.com/land_patterns/90-0063.PDF (T1433-2C)'
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:777]: This is the wrong datasheet when using the generator. After all the suggested land pattern is completely ignored.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 9]
```diff
@@ -773,6 +772,44 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+  body_size_x: 3 +/-0.1
+  body_size_y: 3 +/-0.1
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:779]: the datsheet uses min/nom/max so use the same here (we don't trust ourself to do calculations for good reason)

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 10]
```diff
@@ -774,9 +774,15 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
-  body_size_y: 3 +/-0.1
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0137.PDF (T1433-2C)'
+  body_size_x:
+    minimum: 2.9
+    nominal: 3.0
+    maximum: 3.1
+  body_size_y:
+    minimum: 2.9
+    nominal: 3.0
+    maximum: 3.1
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:785]: Its a bit strange that you mix dimension stile in the same component. Could you either use exclusively the string style or dict style?

**Comment by poeschlr** [comment]
Thanks

**Pull request merged**